prog/install_rhizome.rb

```diff
@@ -22,12 +24,17 @@ class Prog::InstallRhizome < Prog::Base
               IO.copy_stream(_1, tf)
             end
           end
+          file_hash_map[file] = Digest::MD5.file(full_path).hexdigest
```

Comment:
While fine for this, any use of MD5 makes automatic checkers go berserk, needs SOC2 documentation, etc. unless the reasons are especially compelling, use an unremarkable digest. I like sha512-224 when I don't have compatibility issues.

Reply:
haven't thought about the auto-checks and other possible headache points about md5, it was just simple and fast without any security concerns. Switched to SHA 512 now (not sha512-224 because I wanted to stick to the stdlib and IMO anything is fine for this use case, but LMK if you think otherwise).

Reply:
I think OpenSSL is not stdlib but stdlib-adjacent, and we have dependencies for it allllll over the place. Main advantage of 512-224: shorter. Secondary advantage: more resistant to length extension attacks, but who cares in this case I guess. See table at https://en.wikipedia.org/wiki/SHA-2#Comparison_of_SHA_functions. Also SHA512 is faster on modern architectures. This is all nerd-golf though relative to the purpose.

```ruby
require 'openssl'

# Create a SHA-512/224 digest object
digest = OpenSSL::Digest.new('SHA512-224')

# Compute the hash of a string
message = "Hello, world!"
hash = digest.digest(message)

# Convert the binary hash to a hexadecimal string
hex_hash = hash.unpack1('H*')

puts hex_hash

```

Reply:
The `openssl` ruby library is stdlib, there are no plans I am aware of to move it from stdlib to bundled gems. I don't think it will ever move to bundled gems, as rubygems depends on it.  That being said, I would recommend using SHA512 over SHA512-224 unless hash length is a significant issue.

Reply:
Because of ubiquity.  If you are worried about length extension attacks, I recommend we use SHA-384.  That should be more common than SHA-512/224.  SHA-256/224 is more resistant to length extension attacks, but SHA-384 is more resistant to collision attacks, while still providing sufficient security against length extension attacks.

Reply:
SHA-384 suffices, then.